// Webezy.io Generated proto DO NOT EDIT
syntax = "proto3";


package webezy.WebezyProxy.v1;



// [webezyio] - The proxy settings global address
message ProxyAddress {
	// [webezyio] - The address for the proxy
	string address = 1;
	// [webezyio] - The proxy address port
	int32 port = 2;
}

// [webezyio] - The proxy stats sink
message ProxyStatsSink {
	// [webezyio] - The same cluster name as specified at the clusters array
	string tcp_cluster_name = 1;
	// [webezyio] - 
	string prefix = 2;
}

// [webezyio] - The proxy listener configs
message ProxyListener {
	// [webezyio] - The listsener name
	string name = 1;
	// [webezyio] - The listener address details
	webezy.WebezyProxy.v1.ProxyAddress address = 2;
}

// [webezyio] - The proxy endpoint config
message ProxyEndpoint {
	// [webezyio] - The endpoint address
	webezy.WebezyProxy.v1.ProxyAddress address = 1;
}

// [webezyio] - The proxy cluster details
message ProxyCluster {
	// [webezyio] - The cluster name
	string name = 1;
	// [webezyio] - The cluster endpoints
	repeated webezy.WebezyProxy.v1.ProxyEndpoint endpoints = 2;
}

// [webezyio] - The main configurations for envoy proxy
message Config {
	// [webezyio] - The admin configurations
	webezy.WebezyProxy.v1.ProxyAddress admin = 1;
	// [webezyio] - The proxy stats sink configurations
	webezy.WebezyProxy.v1.ProxyStatsSink stats = 2;
	// [webezyio] - The proxy listeners
	repeated webezy.WebezyProxy.v1.ProxyListener listeners = 3;
	// [webezyio] - The proxy clusters
	repeated webezy.WebezyProxy.v1.ProxyCluster clusters = 4;
}

